velox/connectors/hive/HiveDataSink.h

```diff
@@ -192,16 +192,96 @@ class HiveDataSink : public DataSink {
 
  private:
   std::unique_ptr<dwrf::Writer> createWriter(
-      const std::shared_ptr<const HiveWriterParameters>& writerParameters);
+      const std::shared_ptr<const HiveWriterParameters>& writerParameters)
+      const;
+
+  // Move the existing writers to the new indices, if the current input causes
+  // the partitionIdGenerator_ to rehash that changes the IDs of the previous
+  // partition values and the corresponding writers.
+  void moveWriters();
+
+  static RowVectorPtr makeWriterInput(
```

Comment:
This method is a one-liner. It might be better to remove it and call "wrap" directly.